lib/dgram.js

```diff
@@ -78,7 +78,7 @@ exports._createSocketHandle = function(address, port, addressType, fd, flags) {
 function Socket(type, listener) {
   EventEmitter.call(this);
 
-  if (typeof type === 'object') {
+  if (type !== null && typeof type === 'object') {
```

Comment:
@addaleax do you have any docs on this? Not saying it's not the case - I don't really know myself, but a quick google search throws nothing, and in the lib codebase both cases seem to be equally prevalent:

https://github.com/nodejs/node/blob/be68b68d4863f0d389cc46fdf6f1cbcd1b241d0a/lib/https.js#L53
https://github.com/nodejs/node/blob/013d76cb734aa6deb4f468395960fb9d985cd8f5/lib/dns.js#L111

vs

https://github.com/nodejs/node/blob/69796320207542a775446146cd68c2d6da0f5cb8/lib/querystring.js#L172
https://github.com/nodejs/node/blob/b7a8a691b4c6ff0f89b5d8dba3184eaef0efc898/lib/events.js#L63

if we can confirm one case is better, this could mean future improvement PRs and more consistency over the codebase